File: packages/ios/Sources/Components/Stack/Stack.figma.swift
import SwiftUI
import Figma

struct KozmosStackConnect: FigmaConnect {
    let component = Stack<AnyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=170-1027"

    @FigmaEnum(
        "Direction",
        mapping: [
            "Column": StackDirection.vertical,
            "Row": StackDirection.horizontal
        ]
    )
    var direction: StackDirection = .vertical

    @FigmaEnum(
        "Gap",
        mapping: [
            "2": "2",
            "4": "4",
            "6": "6"
        ]
    )
    var gap: String = "2"

    var body: some View {
        Stack(direction: self.direction, spacing: self.spacing) {
            AnyView(Text("Stack item"))
        }
    }

    private var spacing: CGFloat {
        switch self.gap {
        case "4": return 16
        case "6": return 24
        default: return 8
        }
    }
}

File: packages/android/src/main/java/com/kozmos/components/Stack/Stack.figma.kt
package com.kozmos.components.stack

import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=170-1027")
class KozmosStackConnect {
    @FigmaProperty(FigmaType.Enum, "Direction")
    val direction: StackDirection = Figma.mapping(
        "Column" to StackDirection.Vertical,
        "Row" to StackDirection.Horizontal
    )

    @FigmaProperty(FigmaType.Enum, "Gap")
    val gap: String = Figma.mapping(
        "2" to "2",
        "4" to "4",
        "6" to "6"
    )

    @Composable
    fun ComponentExample() {
        KozmosStack(
            direction = direction,
            spacing = gapSpacing()
        ) {
            Text("Stack item")
            Text("Stack item")
        }
    }

    private fun gapSpacing(): Dp = when (gap) {
        "4" -> 16.dp
        "6" -> 24.dp
        else -> 8.dp
    }
}
